src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -147,5 +147,47 @@ public IDataView FilterByKeyColumnFraction(IDataView input, string columnName, d
                 throw Environment.ExceptSchemaMismatch(nameof(columnName), "filter", columnName, "KeyType", type.ToString());
             return new RangeFilter(Environment, input, columnName, lowerBound, upperBound, false);
         }
+
+        /// <summary>
+        /// Shuffle the rows of <paramref name="input"/>.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Shuffle"/> will shuffle the rows of any input <see cref="IDataView"/> by using a streaming pool.
+        /// A pool of <paramref name="poolRows"/> rows will be constructed from the first <paramref name="poolRows"/> rows
+        /// in the dataset. Rows will then be randomly yielded from the pool and replaced with the next row in the dataset
+        /// until all the rows have been yielded.
+        /// If the <paramref name="input"/> is shufflable, then it will read into the pool with a shuffled cursor.
```

Comment:
>/// If the <paramref name="input"/> is shufflable, then it will read into the pool with a shuffled cursor. [](start = 8, length = 106)

I am sorry but what does it mean for the input to be shufflable? When would it not be? Also it might be best not to talk about cursors unless it is necessary I am afraid it might lead to quite a bit of confusion for people who are used to thinking of numpy arrays.

Reply:
I made this more explicit by calling out the IDV CanShuffle property. The docs will now link to IDV and CanShuffle.

---
In reply to: [254189367](https://github.com/dotnet/machinelearning/pull/2427#discussion_r254189367) [](ancestors = 254189367)